### PR TITLE
:mortar_board: Topic25 Define sorting

### PR DESCRIPTION
**Pull request by jameshughes89**
### What
Add a section for defining sorting. It's not overly formal, but it does go into the fact that the elements are the things that have the orderness property, not the algo or the collection. 

### Why
It helps to slow down and think about what sorting is in more detail. Sure, they all know what sorting is, but being deliberate helps. 

### Additional Notes
The point on orderdness is already discussed in ordered bag and binary search tree, so this is just a reminder.

### CODE REVIEW

File: src/site/topic25.rst [hunk 1]
```diff
@@ -18,9 +18,55 @@ Topic #25 --- Sorting
...
+
+* The goal is to arrange the collection of *things* such that they are *in order*
+
+
+Order
+-----
+
+* What *in order* means however will depend on what is being sorted
+* For example
+    * Numbers come with an intrinsic ordering
```
Comment (twentylemon) [inline src/site/topic25.rst:39]: Java uses [natural ordering](https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/Comparable.html) to describe this.

> This ordering is referred to as the class's _natural ordering_, and the class's `compareTo` method is referred to as its _natural comparison method_.

File: src/site/topic25.rst [hunk 2]
```diff
@@ -18,9 +18,55 @@ Topic #25 --- Sorting
...
+    * Strings could be ordered alphabetically
+        * Perhaps by length?
+        * Maybe based on the character's ascii values?
+        * Treat the strings as base 26 integers?
```
Comment (twentylemon) [inline src/site/topic25.rst:43]: This go a little out there a little too quickly. Stick with simple examples like alphabetically, case-insensitive alphabetical, length. The other objects (people, car parts) already dive into weird sorting properties.

File: src/site/topic25.rst [hunk 3]
```diff
@@ -18,9 +18,55 @@ Topic #25 --- Sorting
...
+        * Volume?
+        * The amount of dust particles on the surface?
+
+* With numbers, we don't feel the need to ask
+* But with other things, we need to have a well defined ordering for the *things*
+    * The things themselves have the order-*ness* property, not a sorting algorithm
+
+
+Examples Here
+-------------
```
Comment (twentylemon) [inline src/site/topic25.rst:59]: ..? This title seems very out of place. I'm not sure what it even means.

File: src/site/topic25.rst [hunk 4]
```diff
@@ -18,9 +18,55 @@ Topic #25 --- Sorting
...
+        * Or month and day of birthday?
+    * Car Parts by mass?
+        * Or production code?
+        * Date they were made?
+        * Volume?
+        * The amount of dust particles on the surface?
+
+* With numbers, we don't feel the need to ask
+* But with other things, we need to have a well defined ordering for the *things*
+    * The things themselves have the order-*ness* property, not a sorting algorithm
```
Comment (twentylemon) [inline src/site/topic25.rst:55]: May be worth bringing up [Comparable](https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/Comparable.html) (define natural ordering, and they've seen it a few times already) and [Comparator](https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/util/Comparator.html) (define an _unnatural_ ordering).

File: src/site/topic25.rst (hunk 3 above)
Comment (twentylemon) [reply src/site/topic25.rst:59]: Eh. I'd probably end up not using a section header for it. Even cramming this all into a `..note::` or two would make sense.

File: src/site/topic25.rst [hunk 5]
```diff
@@ -18,9 +18,55 @@ Topic #25 --- Sorting
...
+
+
+Examples Here
+-------------
+
+* For ease, we will use numbers in the examples, but the ideas are generalizable to other *things* if they have a defined order
+
+* We will define a collection of numbers ordered if and only if the elements are in increasing order
```
Comment (twentylemon) [inline src/site/topic25.rst:63]: increasing -> non-decreasing

File: src/site/topic25.rst (hunk 4 above)
Comment (jameshughes89) [reply src/site/topic25.rst:55]: mostly copy/paste

**Pull request merged**